### ISSUE TITLE
Unable to overwrite the aria label value for dialog WAI-ARIA widgets must have an accessible name specified with aria-label, aria-labelledby or the widget's inner text.

### ISSUE THREAD

**Issue by reeteshsingh93** [IBM/carbon-components-react#1932]
## Environment

> Operating system

> Browser

> Automated testing tool and ruleset

> Assistive technology used to verify

## Detailed description

> What version of the Carbon Design System are you using?

  "carbon-components": "8.23.1",
    "carbon-components-react": "5.32.0",

> What did you expect to happen?

aria label value should be present

> What happened instead?

> What WCAG 2.1 checkpoint does the issue violate?

## Steps to reproduce the issue

1. Step one
2. Step two
3. Step three
4. etc.

> Please create a reduced test case in CodeSandbox
> https://codesandbox.io/s/github/IBM/carbon-components-react/tree/master/examples/codesandbox

## Additional information

- Screenshots or code
- Notes

**Comment by dakahn** [comment]
Hello. I need more information to be able to help.
1. What specific component are you referring to here? In the example you provided we've got a button, but the issue seems to be more general
2. Is this throwing an automated testing error for you? If so could you reference that error here. Providing a link to the detailed description would be great. I'll also need the rule sets you're testing against if applicable (like in the case of DAP).

Just tested the link you posted with the latest rule set of DAP and found no errors.

**Comment by reeteshsingh93** [comment]
I am referring to Modal carbon  component here.
We have got a AVT/GVT defect from accessibility testers. They are using IBM dap tool to scan the pages. Which is throwing the above error on a modal.

**Comment by dakahn** [comment]
So if I have this right, you're attempting to set aria-label on a modal, but can't. Does your modal have label and is linking to this label's id in aria-labelledby insufficient?

**Comment by reeteshsingh93** [comment]
We need to add a aria label value for div element here with dialog role. which comes from Modal carbon component.

![image](https://user-images.githubusercontent.com/46310374/53729808-441bbe80-3e9c-11e9-8185-e15a83897228.png)

**Comment by dakahn** [comment]
@joshblack

**Comment by asudoh** [comment]
`<Modal>` has https://github.com/IBM/carbon-components-react/blob/v6.95.0/src/components/Modal/Modal.js#L311 - Are you looking at `<ComposedModal>` @reeteshsingh93?

**Comment by asudoh** [comment]
CC @emyarod for component consistency topic

**Comment by carmacleod** [comment]
Why does `<Modal>` have `modalLabel` and `modalAriaLabel`? Carbon should be hooking up the `aria-label` (or `aria-labelledby`) to the `modalLabel` so that authors don't need to provide it separately.

Also, it would be nice if Carbon could set the `aria-describedby` to point to the element containing the `bx-modal-content__text`. (would need to give the content__text element a unique id)

Even more ideally, it would be nice to include the modal heading with the content in the `aria-describedby`, so that the screen reader reads it as well. (but for that, you would need to restructure your Modal a bit so that the header element and the content element have the same parent).

When a dialog becomes visible and a descendant is focused, a screen reader would typically say the aria-label, followed by the word "dialog" (and possibly "modal"), followed by the text content of the aria-describedby element, and then the focused element's label.

![image](https://user-images.githubusercontent.com/3331913/55297469-508c2c00-53f4-11e9-8bd1-1225c401792a.png)

**Comment by reeteshsingh93** [comment]
@asudoh Which version we can upgrade to get this changes?

**Comment by carmacleod** [comment]
Hmmm, very interesting. I just now looked at the [Vanilla Carbon Modal](https://www.carbondesignsystem.com/components/modal/code), and it already does connect up `aria-labelledby` and `aria-describedby`. Nice.

A couple of surprises, though:
1. I did not know that the Carbon "dialog label" was optional? That is very strange. So, is the "dialog heading" expected to be the dialog title, then? Usually, the text at the top of the dialog is considered to be the dialog title, i.e. "Find" dialog, "Open File" dialog, "Create New User" dialog, "Login" dialog. There must always be a valid `aria-label` on a dialog (so that screen readers can announce the dialog's name when it becomes visible), so either make the Carbon "dialog label" required, or use the Carbon "dialog heading" as the `aria-label`.
2. Usually, the `aria-describedby` points to the sentence or sentences that describe what the dialog is for, or how to use the dialog, i.e. the "Lorem ipsum" text that Carbon has in the "bx--modal-content" element.

So, while the Vanilla Carbon Modal has done a much better job than the React Carbon Modal of hooking up the ARIA label and description, it has some oddities.

I would suggest that any redesign includes thinking about how a Modal should be marked up for accessibility. Note that:
a) `aria-label[ledby]` is not optional. If a Carbon Modal does not have a "label", then `aria-labelledby` needs to point to the "heading".
b) `aria-describedby` is optional, but if there is [a heading and] some text that describes how to use the dialog, then `aria-describedby` should point to it.

**Comment by dakahn** [comment]
@carmacleod Would you be down to open up a ticket with these comments. They're really good and I want to make sure and capture them.

**Comment by reeteshsingh93** [comment]
@carmacleod We have got the following defect from accessibility team:

 The error persists though aria label value is added to its parent element.
![image](https://user-images.githubusercontent.com/46310374/55606420-51c3ad00-5796-11e9-9f6a-e89f2b720c2c.png)
The dialog is with violation because of its role, please refer the info below to report the issue.

![image](https://user-images.githubusercontent.com/46310374/55606432-5ab47e80-5796-11e9-8f2d-ba98bcb3bc78.png)

When can we get this aria label added on dialog role.

**Comment by carmacleod** [comment]
Hi @reeteshsingh93 
Typo in `aria-lavelledby` - there is a `v` where there should be a `b`.  :)

**Comment by carmacleod** [comment]
@dakahn I opened Vanilla Modal issue https://github.com/IBM/carbon-components/issues/2270

**Comment by dakahn** [comment]
Closing this one up as resolved. We can re-open if we need to just holler 🤠
